tests/test_core.py

```diff
@@ -1309,3 +1315,15 @@ def test_DatalakeBadOffsetExceptionRecovery(azure):
     assert azure.cat(a) == data*2
     _put_data_with_retry(azure.azure, 'APPEND', a, data=data)
     assert azure.cat(a) == data*3
+
+
+def test_file_creation_open(azure):
+    with azure_teardown(azure):
+        if azure.exists(a):
+            azure.rm(a)
+        assert not azure.exists(a)
+        f = azure.open(a, "wb")
+        assert azure.exists(a)
+        f.close()
```

Comment:
can you write something with data flag and then do close?

Reply:
That is already done in previous tests. I was wrong on my earlier assumption that flush would not triggering the actual close for some files, as buffer is never empty except on first write operation, which was handled as a special case. 

I was artificially setting the buffer during my own code so that is why it was triggering. This wouldn't happen in a normal operation. 

This test confirms the change that the file is created at open, and not when the flush from close function is called.